### PR TITLE
[Main] Expand Windows test coverage in nodescaling and provisioning

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

We do not have a great coverage of Windows-based testing currently in the framework. While not as prevalent, it is still important that we test these types of clusters.
 
## Solution

Added additional test cases for Windows-based clusters.
 
### Automated Testing
Jenkins jobs will be provided to reviewers offline.

### CODE REVIEW

File: tests/v2/validation/nodescaling/scaling_node_driver_test.go
Comment (slickwarren) [inline tests/v2/validation/nodescaling/scaling_node_driver_test.go:116]: assuming this can be removed instead of commented out

File: tests/v2/validation/nodescaling/scaling_nodepools.go
Comment (slickwarren) [inline tests/v2/validation/nodescaling/scaling_nodepools.go:30]: I think you mentioned this offline but why is this needed for windows but not other node types?

File: go.mod
Comment (slickwarren) [inline go.mod:18]: blocking on this to prevent an accidental merge

**Review by slickwarren — COMMENTED** [review]
overall lgtm; can give full signoff once jenkins jobs are provided + shepherd is merged

File: tests/v2/validation/nodescaling/scaling_node_driver_test.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scaling_node_driver_test.go:116]: That was an accident I will uncomment it out.

File: tests/v2/validation/nodescaling/scaling_nodepools.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scaling_nodepools.go:30]: Yeah, Windows needs to specifically have the syntax shown in the if condition. If I try to do the scale down syntax that Linux nodes do, it doesn't work. Vice versa, it's the same story.

**Review by susesgartner — APPROVED** [review]
LGTM

**Review by Josh-Diamond — APPROVED** [review]

**Review by Josh-Diamond — APPROVED** [review]

File: go.mod
Comment (slickwarren) [inline go.mod:19]: blocking on this as a formality

File: tests/v2/actions/machinepools/machinepools.go
Comment (slickwarren) [inline tests/v2/actions/machinepools/machinepools.go:63]: if you have time to add a comment justifying the double for loop, that would be nice to have here.

**Review by slickwarren — CHANGES_REQUESTED** [review]
lgtm

**Review by susesgartner — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]

**Pull request merged**
